modules/sc-mesh-secure-deployment/src/nats/cbma/secure_socket/client.py

```diff
@@ -41,7 +41,7 @@ def __init__(self,
         self.server_port = server_port
         self.macsec_callback = macsec_callback
 
-        super().__init__(certificates, SSL.TLSv1_2_METHOD)
+        super().__init__(certificates, SSL.SSLv23_METHOD)
```

Comment:
I think this will work, although it has deprecation warning.

The newer way is to pass SSL.TLS_CLIENT_METHOD and set minimum version by ctx options.

Reply:
Yes, that's right, but this method and the `set_min_proto_version` are only available from PyOpenSSL 21 and we only have 19.1 in our comms devices.

For completeness, CBMA `old-requirements-fixes` branch is the one being pushed to mesh_com with this deprecated code, but `develop` branch already use the new code:
- https://github.com/tiiuae/cbma/blob/develop/secure_socket/client.py#L44
- https://github.com/tiiuae/cbma/blob/develop/secure_socket/secure_context.py#L46

:star2: